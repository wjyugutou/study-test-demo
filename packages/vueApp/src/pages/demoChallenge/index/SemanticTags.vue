<script lang='ts' setup>
defineOptions({
  description: '语义化标签',
})
</script>

<template>
  <div flex items-center justify-center>
    <code text-26px>{{ '<sub></sub>' }}</code>
    <span mr-2>:</span>
    <p>
      水的化学式: H<sub>2</sub>O
    </p>
  </div>
  <div flex items-center justify-center>
    <code text-26px>{{ '<sup></sup>' }}</code>
    <span mr-2>:</span>
    <p>2 <sup>2</sup> = 4 </p>
  </div>

  <details>
    details标签
  </details>
</template>
