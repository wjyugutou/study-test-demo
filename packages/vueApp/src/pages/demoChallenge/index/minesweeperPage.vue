<script lang='ts' setup>
</script>

<template>
  <Minesweeper />
</template>
