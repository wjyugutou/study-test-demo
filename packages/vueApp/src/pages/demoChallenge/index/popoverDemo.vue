<script lang='ts' setup>
defineOptions({
  name: 'Popover',
  label: 'Popover',
})
</script>

<template>
  <div w-800px h-400px overflow-auto pl-200px m-auto mt-20 flex border="1px solid red">
    <div bg-pink w-200px h-1000px>
      <Popover placement="top" strategy="absolute" mode="click" content="String content">
        <div w="200px" bg="green-400" dark:bg="green-800">
          strategy="absolute" click
        </div>
        <template #content>
          <div>
            asdasd
            <div>content1</div>
            <div>content2</div>
            <div>content3</div>
          </div>
        </template>
      </Popover>
    </div>
    <div bg-blue-400 w-200px h-1000px>
      <Popover placement="top" strategy="fixed" mode="hover" content="String content">
        <div w="200px" bg="green-400" dark:bg="green-800">
          strategy="fixed" hover
        </div>
        <template #content>
          <div>
            asdasd
            <div>content1</div>
            <div>content2</div>
            <div>content3</div>
          </div>
        </template>
      </Popover>
    </div>
  </div>
</template>

<style scoped>

</style>
