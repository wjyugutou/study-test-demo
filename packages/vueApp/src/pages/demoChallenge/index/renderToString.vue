<script lang='ts' setup>
import { renderToString } from 'vue/server-renderer'
import { createSSRApp } from 'vue'
import Basic from '@/components/ssrComponents/Basic.vue'

defineOptions({
  description: 'vue: createSSRApp,renderToString 方法',
})

const divRef = ref<HTMLDivElement>()
const app = createSSRApp(Basic)

async function init() {
  const result = await renderToString(app)
  console.log(result)
  divRef.value!.innerHTML = result
}

onMounted(() => {
  init()
})
</script>

<template>
  aaa
  <div ref="divRef" />
  <Basic />
</template>
