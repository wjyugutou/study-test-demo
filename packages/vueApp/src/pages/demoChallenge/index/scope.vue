<script lang='ts' setup>
import type { CSSProperties } from 'vue'

defineOptions({
  description: '炫彩边框 动画',
})

const box = ref() as Ref<HTMLDivElement>
const dragStyle = reactive<CSSProperties>({
  top: '0px',
  left: '0px',
  transform: 'translateZ(0px)',
})
const contentStyle = reactive<CSSProperties>({
  top: '0px',
  left: '0px',
})

function pointerdown(de: PointerEvent) {
  let initX = de.x
  let initY = de.y
  const maxX = window.innerWidth - box.value!.clientWidth
  const maxY = parent.innerHeight - box.value!.clientHeight;

  (de.target as Element).setPointerCapture(de.pointerId)

  const stopMove = useEventListener(box, 'mousemove', (e: MouseEvent) => {
    e.preventDefault()
    const curX = e.x
    const curY = e.y
    const disX = curX - initX
    const disY = curY - initY
    initX = e.x
    initY = e.y
    const dragTop = Number.parseInt(`${dragStyle.top!}`) + disY
    const dragLeft = Number.parseInt(`${dragStyle.left!}`) + disX

    if (dragLeft < 0 || dragLeft > maxX || dragTop < 0 || dragTop > maxY)
      return

    dragStyle.top = `${dragTop}px`
    dragStyle.left = `${dragLeft}px`
    contentStyle.top = `${Number.parseInt(`${contentStyle.top!}`) - disY}px`
    contentStyle.left = `${Number.parseInt(`${contentStyle.left!}`) - disX}px`
  })

  const stopLeave = useEventListener(box, 'pointerleave', (e: MouseEvent) => {
    stopMove()
    stopLeave();
    (de.target as Element).releasePointerCapture(de.pointerId)
  })
  const stopUp = useEventListener(box, 'pointerup', (e: MouseEvent) => {
    stopMove()
    stopUp();
    (de.target as Element).releasePointerCapture(de.pointerId)
  })
}

useDrag(box)
</script>

<template>
  <div ref="box" absolute h-100 w-100 select-none overflow-hidden :style="dragStyle" @pointerdown="pointerdown">
    <div class="inner" relative h-1080px w-1920px p-10 :style="contentStyle">
      <h3 class="mb-2 mt-4 font-bold">
        The standard Lorem Ipsum passage, used since the 1500s
      </h3><p>"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."</p><h3 class="mb-2 mt-4 font-bold">
        Section 1.10.32 of "de Finibus Bonorum et Malorum", written by Cicero in 45 BC
      </h3><p>"Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt. Neque porro quisquam est, qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit, sed quia non numquam eius modi tempora incidunt ut labore et dolore magnam aliquam quaerat voluptatem. Ut enim ad minima veniam, quis nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid ex ea commodi consequatur? Quis autem vel eum iure reprehenderit qui in ea voluptate velit esse quam nihil molestiae consequatur, vel illum qui dolorem eum fugiat quo voluptas nulla pariatur?"</p><h3 class="mb-2 mt-4 font-bold">
        1914 translation by H. Rackham
      </h3><p>"But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness. No one rejects, dislikes, or avoids pleasure itself, because it is pleasure, but because those who do not know how to pursue pleasure rationally encounter consequences that are extremely painful. Nor again is there anyone who loves or pursues or desires to obtain pain of itself, because it is pain, but because occasionally circumstances occur in which toil and pain can procure him some great pleasure. To take a trivial example, which of us ever undertakes laborious physical exercise, except to obtain some advantage from it? But who has any right to find fault with a man who chooses to enjoy a pleasure that has no annoying consequences, or one who avoids a pain that produces no resultant pleasure?"</p><h3 class="mb-2 mt-4 font-bold">
        Section 1.10.33 of "de Finibus Bonorum et Malorum", written by Cicero in 45 BC
      </h3><p>"At vero eos et accusamus et iusto odio dignissimos ducimus qui blanditiis praesentium voluptatum deleniti atque corrupti quos dolores et quas molestias excepturi sint occaecati cupiditate non provident, similique sunt in culpa qui officia deserunt mollitia animi, id est laborum et dolorum fuga. Et harum quidem rerum facilis est et expedita distinctio. Nam libero tempore, cum soluta nobis est eligendi optio cumque nihil impedit quo minus id quod maxime placeat facere possimus, omnis voluptas assumenda est, omnis dolor repellendus. Temporibus autem quibusdam et aut officiis debitis aut rerum necessitatibus saepe eveniet ut et voluptates repudiandae sint et molestiae non recusandae. Itaque earum rerum hic tenetur a sapiente delectus, ut aut reiciendis voluptatibus maiores alias consequatur aut perferendis doloribus asperiores repellat."</p><h3 class="mb-2 mt-4 font-bold">
        1914 translation by H. Rackham
      </h3><p>"On the other hand, we denounce with righteous indignation and dislike men who are so beguiled and demoralized by the charms of pleasure of the moment, so blinded by desire, that they cannot foresee the pain and trouble that are bound to ensue; and equal blame belongs to those who fail in their duty through weakness of will, which is the same as saying through shrinking from toil and pain. These cases are perfectly simple and easy to distinguish. In a free hour, when our power of choice is untrammelled and when nothing prevents our being able to do what we like best, every pleasure is to be welcomed and every pain avoided. But in certain circumstances and owing to the claims of duty or the obligations of business it will frequently occur that pleasures have to be repudiated and annoyances accepted. The wise man therefore always holds in these matters to this principle of selection: he rejects pleasures to secure other greater pleasures, or else he endures pains to avoid worse pains."</p>
    </div>
    <div class="clip" />
  </div>
</template>

<style scoped>
.inner  {
  pointer-events: none;
}

.clip {
  position: absolute;
  top: 0;
  left: 0;
  border: 2px solid;
  border-image-source: linear-gradient(45deg, rgb(184 159 218), rgb(15 48 194));
  border-image-slice: 1;
  width: 100%;
  height: 100%;
  animation: clip-border 2s infinite;
  animation-timing-function: linear;

  /* 边框位置 */
  clip-path: inset(0 90% 0 0);
}

@keyframes clip-border {
  0%,
  100% {
    clip-path: inset(90% 0 0 0);
    filter: hue-rotate(0deg);
  }

  25% {
    clip-path: inset(0 90% 0 0);
  }

  50% {
    clip-path: inset(0 0 90% 0);
    filter: hue-rotate(360deg);
  }

  75% {
    clip-path: inset(0 0 0 90%);
  }

}
</style>
