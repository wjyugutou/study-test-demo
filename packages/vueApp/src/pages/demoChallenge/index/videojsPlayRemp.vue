<script lang='ts' setup>
defineOptions({ description: 'videojs demo' })

const player = ref()

function initVideoPlayer() {

}

onMounted(() => {
  initVideoPlayer()
})
</script>

<template>
  <video id="video" class="vjs-big-play-centered" src="" />
</template>
