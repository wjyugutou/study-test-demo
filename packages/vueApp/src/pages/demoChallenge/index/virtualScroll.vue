<script lang='ts' setup>
import { random } from 'lodash-es'

defineOptions({ description: '虚拟列表' })

const data = Array.from({ length: 1000 }).map((item, index) => ({ content: `但是价格法大师傅--${index}`, index, src: '', height: random(20, 165) }))
</script>

<template>
  <div>
    <h1 text-center>
      动态高度
    </h1>
    <VirtualList :data-source="data" class="virtualList">
      <template #default="{ record, index } ">
        <div :style="{ height: `${record.height}px` }" :class="{ 'bg-pink-400': record.index % 2 === 0, 'bg-blue-400': record.index % 2 !== 0 }">
          {{ record }}--          {{ index }}
        </div>
      </template>
    </VirtualList>

    <CssVirtualList h-200px>
      <div v-for="(item) of data" :key="item.index">
        <p>5asdasdas <br> 5</p>
      </div>
    </CssVirtualList>
  </div>
</template>

<style>
.virtualList {
  overflow-y: auto;
  position: relative;
  height: 300px;
}
</style>
