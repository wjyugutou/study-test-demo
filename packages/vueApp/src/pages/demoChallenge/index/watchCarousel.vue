<script lang='ts' setup>
import img from '@/static/images/mio.jpg'
import img1 from '@/static/images/1.jpg'

defineOptions({ description: '手表型轮播图' })

const carousel = ref() as Ref<HTMLDivElement>
const items = ref() as Ref<HTMLDivElement[]>

const count = ref(22)
// 半径
const radius = ref(0)
// 每个子元素角度
const theta = 360 / count.value

const index = ref(0)
function rotateFunc() {
  const angel = theta * index.value * -1

  carousel.value.style.transform = `translateZ(${-radius.value}px) rotateX(${-Math.round(angel)}deg)`
}

function initial() {
  radius.value = Math.round((carousel.value.parentElement!.clientWidth / 1.8) / (Math.tan(Math.PI / count.value * 2)))

  items.value.forEach((li, index) => {
    const angel = theta * index

    // 先rotate translateZ的位移方向会变化从而实现效果
    li.style.transform = `rotateX(${-Math.round(angel)}deg) translateZ(${radius.value}px)`
  })
  rotateFunc()
}

function changeActiveUp() {
  index.value++
  rotateFunc()
}
function changeActiveDown() {
  index.value--
  rotateFunc()
}

onMounted(initial)
</script>

<template>
  <div relative m-auto w-500px flex items-center justify-center :style="{ height: `${radius}px` }">
    <div transition="~ 1s" relative h-500px w-500px perspective-300px>
      <ol ref="carousel" absolute h-210px w-full preserve-3d transition="~">
        <template v-for="i in count" :key="i">
          <li ref="items" border="~ 2px gray-400" absolute h-200px w-500px overflow-hidden b-rd-4 op-70>
            <img pointer-events-none w-full translate-y--10px :src="i % 2 === 0 ? img : img1" alt="">
          </li>
        </template>
      </ol>
    </div>
    <div class="arrows" absolute top="50%" right-10px translate-y="-50%" flex flex-col gap-5>
      <button border="~ gray-400" bg="gray-400/50" transform-rotate-180deg transition hover:bg-gray-400 @click="changeActiveUp">
        <div i-carbon-caret-down text-8 />
      </button>
      <button border="~ gray-400" bg="gray-400/50" transition hover:bg-gray-400 @click="changeActiveDown">
        <div i-carbon-caret-down text-8 />
      </button>
    </div>
  </div>
</template>
