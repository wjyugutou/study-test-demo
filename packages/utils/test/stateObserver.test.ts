import { expect, test, vi } from 'vitest'
import { stateObserver } from '../index'

const so = stateObserver({ name: 'stateObserver', age: 14 })
so.on('nameUpdate', () => {})

function ageUpdate(oldValue: number, newValue: number) {
  console.log('ageUpdateFn', { oldValue, newValue })
}

so.on('ageUpdate', ageUpdate)

so.on('ageUpdate', () => {
  console.log('ageUpdate 222')
})

test('stateObserver updateFn', () => {
  const spyF = vi.fn(ageUpdate)

  so.state.age = 111
  so.state.age = 222
  so.state.age = 333
  console.log('getMockName', spyF.getMockName())

  expect(spyF).toHaveBeenCalled()

  // expect(so.state.age = 20).toMatchSnapshot()
})
