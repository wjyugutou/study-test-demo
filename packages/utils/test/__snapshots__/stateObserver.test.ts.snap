// Vitest Snapshot v1

exports[`stateObserver updateFn 1`] = `undefined`;

exports[`stateObserver updateFn 2`] = `1`;
